### PR TITLE
Bubble sort algorithm

### PR DESCRIPTION
**Pull request by 8Bts**
In this project we implemented bubble sort algorithm, so we created two ruby methods called bubble_sort and bubble_sort_by. The First method requires an Integer array as a parameter. The second method also requires an Integer array but you also need to pass a block which will be used to sort an array.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,50 @@
...
+![Bubble Sort Animation](Bubble-sort-example.gif)
+
+The purpose of the project is to implement Bubble sort algorithm with Ruby using yield and passing a block to a method. 
+
+## Development
+
+* Run script
+```
+ruby bubble_sort.rb
+```
```
Comment (shshamim63) [inline README.md]: [Optional]
Replace line 12 with the following line. It will give you a fancy look.
```
```ruby
```
```

File: README.md [hunk 2]
```diff
@@ -1 +1,50 @@
-# Bubble-Sort
\ No newline at end of file
+# Bubble Sort
+
+This is a collaborative project about building a bubble sort algorithm.
+
+![Bubble Sort Animation](Bubble-sort-example.gif)
```
Comment (shshamim63) [inline README.md]: This animation is not rendering. It seems not included in the repo or the link is invalid.

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        flag = true
+      end
+      k += 1
+    end
+    break if flag == false
+
+    i += 1
+  end
+  arr
+end
```
Comment (shshamim63) [inline main.rb]: I tried to execute the method with an unsorted array. But did not receive the expected result.
```ruby
p bubble_sort([9,2,1,6,5,4,8,10,7,3]) # The result I got [2, 1, 6, 5, 4, 8, 9, 7, 3, 10]
```

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Required Changes :tractor: 
___
Hi @8Bts and @chindrop :wave: 
Nice initiative, well-indented code. But there is a major issue that is blocking me to approve this right now. Let's take down the issue.
___
### :spades: Mandatory Changes :spades: 
___
- The bubble_sort method is not returning a properly sorted array. Check out the comment attached to the code block.
- The gif you have added is not working. Please fix it.
___
#### Please resolve the above-mentioned issues and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by 8Bts** [comment]
Thanks for review @shshamim63. We fixed problems mentioned by you.

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**